src/auro-input.js

```diff
@@ -106,16 +126,141 @@ export default class AuroInput extends BaseInput {
     return null
   }
 
+  connectedCallback() {
+    super.connectedCallback();
+
+    // Process auto-formating if defined
+    if (this.type) {
+      let config = null;
+
+      // Process only valid format strings
+      switch (this.type) {
+        case 'credit-card':
+          config = {
+            creditCard: true
+          }
+
+          this.numericKeyboard = true;
+
+          break;
+        // add additional supported formats and their config JSON here
+        default:
+          // Do nothing
+      }
+
+      // initialize CleaveJS if we have a defined config for the requested format
+      if (config) {
+        // eslint-disable-next-line no-unused-vars
+        const cleave = new Cleave(this, config);
+      }
+    }
+  }
+
+  /**
+   * @private function to return html for credit card icon
+   * @returns {string} html string
+   */
+  processCreditCard() {
+    const card = this.matchInputValueToCreditCard();
+
+    this.maxLength = card.formatLength;
+
+    if (this.icon) {
+      const svg = new DOMParser().parseFromString(card.cardIcon.svg, 'text/html').body.firstChild;
+
+      return html`
+      <div class="creditCard-icon">
+        ${svg}
+      </div>
+      `
+    }
+
+    return html``;
+  }
+
+  /**
+   * @private function returns data object for credit card matching value comparison
+   * @returns {object} JSON with data for credit card formatting
+   */
+  matchInputValueToCreditCard() {
```

Comment:
Does Cleave have a way to validate if all the numbers have been entered? Just noticed that there is no validation check being done on blur from the input. 

If Cleave can support this, is there a message that can be placed in the `error` attribute?

Reply:
CleaveJS does not handle any form of validation - purely formatting. We would need to extend our own validation rules here.

This is good thinking, @blackfalcon. I'll get a check for that baked in.